src/metabase/task/send_pulses.clj

```diff
@@ -1,136 +1,195 @@
+    (doseq [[{:keys [pulse_id] :as schedule-map} pc-ids] trigger-slot->pc-ids]
+      (update-send-pulse-trigger-if-needed! pulse_id schedule-map :add-pc-ids (set pc-ids)))))
+
+(jobs/defjob ^{:doc "Reprioritizes SendPulse jobs based on the PulseChannels that are scheduled to run at the same time."}
+  RePrioritizeSendPulses
+  [_job-context]
+  (reprioritize-send-pulses!))
+
+;;; -------------------------------------------------- Task init ------------------------------------------------
 
 (defmethod task/init! ::SendPulses [_]
-  (let [job     (jobs/build
-                 (jobs/of-type SendPulses)
-                 (jobs/with-identity (jobs/key send-pulses-job-key)))
-        trigger (triggers/build
-                 (triggers/with-identity (triggers/key send-pulses-trigger-key))
-                 (triggers/start-now)
-                 (triggers/with-schedule
-                   (cron/schedule
-                    ;; run at the top of every hour
-                    (cron/cron-schedule "0 0 * * * ? *")
-                    ;; If send-pulses! misfires, don't try to re-send all the misfired Pulses. Retry only the most
-                    ;; recent misfire, discarding all others. This should hopefully cover cases where a misfire
-                    ;; happens while the system is still running; if the system goes down for an extended period of
-                    ;; time we don't want to re-send tons of (possibly duplicate) Pulses.
-                    ;;
-                    ;; See https://www.nurkiewicz.com/2012/04/quartz-scheduler-misfire-instructions.html
-                    (cron/with-misfire-handling-instruction-fire-and-proceed))))]
-    (task/schedule-task! job trigger)))
+  (let [send-pulse-job       (jobs/build
+                              (jobs/with-identity send-pulse-job-key)
+                              (jobs/with-description "Send Pulse")
+                              (jobs/of-type SendPulse)
+                              (jobs/store-durably))
+        re-proritize-job     (jobs/build
+                              (jobs/with-identity reprioritize-send-pulse-job-key)
+                              (jobs/with-description  "Update SendPulses Pjiority")
+                              (jobs/of-type RePrioritizeSendPulses)
+                              (jobs/store-durably))
+        re-proritize-trigger (triggers/build
+                              (triggers/with-identity reprioritize-send-pulse-trigger-key)
+                              (triggers/for-job reprioritize-send-pulse-job-key)
+                              (triggers/start-now)
+                              (triggers/with-schedule
+                                (cron/schedule
+                                 (cron/cron-schedule "0 0 1 ? * 7 *") ; at 1am on Saturday every week
+                                 (cron/with-misfire-handling-instruction-ignore-misfires))))]
```

Comment:
Same problem with misfire policy as above. We want this to fire the most recent misfire, discarding all others.

Reply:
7938b442f3